### ISSUE TITLE
Update `@yamada-ui/accrodion` accessibility

### ISSUE THREAD

**Issue by nakayan5** [yamada-ui/yamada-ui#1676]
### Description

his issue aims to address accessibility deficiencies in our accrodion component as per W3C guidelines.
The component currently does not meet the accessibility standards outlined in the W3C documentation.

The goal is to ensure the component is accessible to all users, including those using assistive technologies.
Ensure that all unchecked items in this checklist should be addressed.

Keyboard Interaction
Enter or Space:
- [x] ~~When focus is on an expanded panel's header, it collapses the panel (if supported).~~

WAI-ARIA Roles, States, and Properties
Accordion Headers:
- [ ]  Each header button is wrapped in an element with role="heading" and an appropriate aria-level.

Heading Elements:
- [ ] The button element is the only element inside the heading element.
- [ ] Visually persistent elements are not included inside the heading element.

Control Association:
- [ ] The header button has aria-controls set to the ID of the element containing the panel content.

Collapsible Panels:
- [ ] If the panel is visible and cannot be collapsed, the header button has aria-disabled="true".

Panel Containers:
- [ ] Each panel content container has role="region" and aria-labelledby set to the button controlling the panel display.
- [ ] Avoid using role="region" if it creates landmark region proliferation (e.g., more than 6 expandable panels).

Nested Structure:
- [ ] Use role="region" if panels contain heading elements or a nested accordion to help screen reader users perceive structure.

### Problem Statement/Justification

later

### Are you willing to participate in implementing this feature and create a pull request include the implementation?

None

### Additional Information

none

**Comment by hirotomoyamada** [comment]
@nakayan5 

> When focus is on an expanded panel's header, it collapses the panel (if supported).

It's already supported if you set `isToggle` to `true` for `Accordion` component.

**Comment by nakayan5** [comment]
@hirotomoyamada 
certainly
